### PR TITLE
fix(sim): Verilator time precision

### LINKED ISSUE #919: the high time precision breaks when old verilator is used.
While using "sbt test" to test my design below message prompt.
``` log
[Progress] Verilator compilation started
../verilator/VAddressExpander__spinalWrapper.cpp: In constructor ‘Wrapper_1::Wrapper_1(const char*)’:
../verilator/VAddressExpander__spinalWrapper.cpp:192:41: error: ‘timeprecision’ is not a member of ‘Verilated’
  192 |       this->time_precision = Verilated::timeprecision();
      |                                         ^~~~~~~~~~~~~
```

My verilator version is: Verilator 4.028 2020-02-06 rev v4.026-92-g890cecc1
This is the default version Ubuntu 20 included.

### PR DESCRIPTION
**Pull request by kleinai**
Would have used VPI to get the time precision, but I couldn't get that to compile correctly. This solution looks the version and generates code accordingly.

Fixes #919

### CODE REVIEW

**Comment by numero-744** [comment]
LGTM if it passes checks

File: sim/src/main/scala/spinal/sim/VerilatorBackend.scala [hunk 1]
```diff
@@ -644,7 +645,7 @@ JNIEXPORT void API JNICALL ${jniPrefix}disableWave_1${uniqueId}
         FileUtils.copyDirectory(workspaceCacheDir, workspaceDir)
       }
 
-      genWrapperCpp()
+      genWrapperCpp(verilatorVersionDeci > BigDecimal("4.034"))
```
Comment (numero-744) [inline sim/src/main/scala/spinal/sim/VerilatorBackend.scala:648]: I'm curious, how have you found this value `4.034`?

Comment (kleinai) [reply sim/src/main/scala/spinal/sim/VerilatorBackend.scala:648]: First tagged commit after the time precision function was added to Verilator

Comment (numero-744) [reply sim/src/main/scala/spinal/sim/VerilatorBackend.scala:648]: So isn't is `>=`?

**Comment by numero-744** [comment]
@Readon do you want to test it before we merge it?

**Comment by Readon** [comment]
This works but #821 changes API which still breaks the existing code.

**Pull request merged**